### DISCUSSION TITLE
Can I use kserve without istio?

### DISCUSSION THREAD

**Discussion by benjaminhuo** [kubeflow/kfserving#2039] [Q&A]
Knative can be installed without istio, there're other alternatives for istio such as kourier, can I use knative + kourier together with kserve? https://knative.dev/docs/install/serving/install-serving-with-yaml/#install-a-networking-layer

**Comment by kabartay** [comment]
What is the status of this question?

**Comment by sivanantha321** [comment]
Yes. You can.https://kserve.github.io/website/docs/admin-guide/serverless/kourier-networking

**Comment by cananoo** [comment]
Yes. KServe can run with Knative + Kourier without Istio.

The important wiring is:

1. Install the Kourier release that matches your Knative version:

       kubectl apply -f https://github.com/knative/net-kourier/releases/download/$KNATIVE_VERSION/kourier.yaml

2. Tell Knative Serving to use Kourier:

       kubectl patch configmap/config-network --namespace knative-serving --type merge --patch '{"data":{"ingress-class":"kourier.ingress.networking.knative.dev"}}'

3. Disable KServe's top-level Istio VirtualService creation:

       kubectl patch configmap/inferenceservice-config -n kserve --type=strategic -p '{"data": {"ingress": "{ \"disableIstioVirtualHost\": true}"}}'

4. Restart the KServe controller:

       kubectl rollout restart deployment kserve-controller-manager -n kserve

For a local test, port-forward the 3scale-kourier-gateway service or pod and use the hostname from .status.url in the Host header when calling the InferenceService. Kourier covers the Knative ingress path, but it does not provide all of Istio's advanced traffic-management features.

Reference: https://kserve.github.io/website/docs/admin-guide/serverless/kourier-networking
